api/institutions/authentication.py

```diff
@@ -102,6 +103,11 @@ def authenticate(self, request):
         # unconfirmed, etc.).
         user, created = get_or_create_user(fullname, username, reset_password=False)
 
+        # The `department` field is updated each login when it was changed.
+        if department and user.department != department:
+            user.department = department
+            user.save()
```

Comment:
As discussed in person, let's move this part to after the status check so that we won't have the department accidentally set on any pending user. This is similar to a problem we had with affiliations.

cc @brianjgeiger